tests/test_structlog_formatter.py

```diff
@@ -22,7 +22,12 @@
 
 
 def make_event_dict():
-    return {"event": "test message", "log.logger": "logger-name"}
+    return {
+        "event": "test message",
+        "log.logger": "logger-name",
+        "foo": "bar",
```

Comment:
Do we really want a key to be thrown away silently instead of erroring in this case?

Reply:
You're right -- error log but still report the line, I guess?

Reply:
Maybe instead of expanding the dotting in these cases we can keep the dots? Unsure if that'd add a bunch of logic to the expanding or be simpler than I think it'd be.

Reply:
Back to erroring. The key was that we needed to fix the handling around `event` and `event.dataset`, both of which will be defined by the python agent. I switched it to rename `event` to `message` before processing the dictionary. Now conflicts will still pop up but at least we handle the expected conflict correctly.